### DISCUSSION TITLE
Two Way ssl not working in k8s : emqx  4.2.13

### DISCUSSION THREAD

**Discussion by and-dzh3** [emqx/emqx#5419] [Q&A]
Hi 
I want setup authentication in emqx broker with cn from self-signed certificate 
i issued for  clients certificate and base on username and certificate cn , give them acl rules where they can publish massages 
running emqx on k8s .
Broker receives trafic after ssl of load on AWS load balancer. 

my Error :
`10.100.10.9:32978 [Channel] Client test (Username: 'test') login failed for not_authorized`

  EMQX_CLUSTER__K8S__ADDRESS_TYPE: "hostname"
  EMQX_CLUSTER__K8S__SUFFIX: "svc.cluster.local"
  EMQX_CLUSTER__K8S__APP_NAME: "emqx"
  EMQX_CLUSTER__K8S__SERVICE_NAME: "emqx"
  EMQX_CLUSTER__K8S__NAMESPACE: "emqx"
  EMQX_ALLOW_ANONYMOUS: "false" ##allow_anonymous
  EMQX_LISTENER__TCP__EXTERNAL__PEER_CERT_AS_USERNAME: cn
  EMQX_LISTENER__SSL__EXTERNAL__PEER_CERT_AS_USERNAME: cn
  EMQX_LISTENER__SSL__EXTERNAL__FAIL_IF_NO_PEER_CERT: true
  EMQX_LISTENER__SSL__EXTERNAL__VERIFY: verify_peer
  EMQX_LISTENER__SSL__EXTERNAL__KEYFILE: /etc/certs/emqx.key
  EMQX_LISTENER__SSL__EXTERNAL__CERTFILE: /etc/certs/emqx.pem
  EMQX_LISTENER__SSL__EXTERNAL__CACERTFILE: /etc/certs/ca.pem

loaded plagins "emqx_recon,emqx_retainer,emqx_management,emqx_dashboard,emqx_statsd"

client certificate generation with ca.pem same as in emqx roker  

openssl x509 -req -in ./client.csr -CA ca.pem -CAkey ca.key -CAcreateserial -out client.pem -days 3650 -sha256
Signature ok
subject=/CN=test
Getting CA Private Key

what I can do for enabling wanted functionality

**Comment by JimmyBjorklund** [comment]
This works for me, i am running 5.3.0

```
listeners.ssl.default {
    ## Port or Address to listen on, 0 means disable
    bind = 8883 ## or with an IP e.g. "127.0.0.1:8883"
    enabled = true
    acceptors = 16
    enable_authn = false
    max_connections = infinity
    mountpoint = ""
    proxy_protocol = false
    proxy_protocol_timeout = 3s
    tcp_options {
        active_n = 100
        backlog = 1024
        buffer = 4KB
        high_watermark = 1MB
        keepalive = none
        nodelay = true
        reuseaddr = true
        send_timeout = 15s
        send_timeout_close = true
    }
    ssl_options {
        cacertfile = "${EMQX_ETC_DIR}/certs/xx/ca.crt"
        certfile = "${EMQX_ETC_DIR}/certs/xx/server.crt"
        keyfile = "${EMQX_ETC_DIR}/certs/xx/server.key"
        verify = verify_peer # to enable
        ## if `verify' is enabled, whit true, the connection fails if the client does not have a certificate to send
        fail_if_no_peer_cert = true

        ## Enable TLS session reuse
        reuse_sessions = true

        ## Maximum number of non-self-issued intermediate certificates that can follow the peer certificate in a valid certification path
        depth = 10

        ## Which versions are to be supported
        versions = [tlsv1.3, tlsv1.2]

        ## TLS cipher suite names
        ## Note: By default, all available suites are supported, you do not need to set this
        ciphers = "TLS_AES_256_GCM_SHA384,TLS_AES_128_GCM_SHA256,ECDHE-RSA-AES256-GCM-SHA384"

        ## Allows a client and a server to renegotiate the parameters of the SSL connection on the fly
        secure_renegotiate = true

        ## Log level for SSL communication
        ## Type: emergency | alert | critical | error | warning | notice | info | debug | none | all
        log_level = notice

        ## Hibernate the SSL process after idling for amount of time reducing its memory footprint
        hibernate_after = 5s

        ## Forces the cipher to be set based on the server-specified order instead of the client-specified order
        honor_cipher_order = true

        ##  Setting this to false to disable client-initiated renegotiation
        client_renegotiation = true

        ## Maximum time duration allowed for the handshake to complete
        handshake_timeout = 15s

    }
}
```
